### PR TITLE
fix: standardizing the quick-start for spring-boot school timetabling

### PR DESCRIPTION
**Pull request by zepfred**
# Description
The pull request updates the spring-boot quick start module for School Timetabling. All domain and controller classes were adapted from the Quarkus project to work with Spring Boot. The major changes are described as follows:
- All REST annotations were updated to match Spring Boot annotations, such as @RestController and @GetMapping.
- A new global error handler named `TimetableSolverExceptionHandler` has been added.
- A new test configuration class named `ConstraintConfig` has been added to allow injection of a bean class `ConstraintVerifier<TimetableConstraintProvider, Timetable>`.
- The Maven and Gradle files were updated and tested.
- A new dependency called `springdoc-openapi-starter-webmvc-ui` has been added to enable the documentation of endpoints.

solver1
solver2
solver3

Issue #182

### CODE REVIEW

File: technology/java-spring-boot/build.gradle
Comment (triceo) [inline technology/java-spring-boot/build.gradle:42]: If you're upgrading versions of these components, please do it consistently with the other School Timetabling quickstart. We want them to be truly identical, except for the places where they can not be for obvious reasons.

(Please remember to test the quickstart's UI after modifying these dependency versions.)

In general, let's try and keep the versions of *all* dependencies (not just this) in sync across quickstarts. Much of this is done automatically through Dependabot, but for some reason, Gradle doesn't work too well with it.

**Review by triceo — CHANGES_REQUESTED** [review]
I only have one comment about dependencies, for consistency.
Also, please update the quickstart's README with the new picture and if the steps have changed, do that too. (Although I don't think they have.)

Once addressed, I'll merge.

Nice work!

File: technology/java-spring-boot/build.gradle
Comment (zepfred) [reply technology/java-spring-boot/build.gradle:42]: The Gradle files have been updated to match the Maven files, and all modules are now using version `5.15.1` of the `font-awesome` library.

I have successfully executed the Gradle commands to build and run the projects `technology/java-spring-boot` and `use-cases/school-timetabling`.

**Review by triceo — APPROVED** [review]
LGTM when CI is green.

**Pull request merged**
